### PR TITLE
clean redirection after unpublish node

### PR DESCRIPTION
**Pull request by veolys-nanne**
https://github.com/open-orchestra/open-orchestra-cms-bundle/pull/1282
https://github.com/open-orchestra/open-orchestra-model-bundle/pull/485
https://github.com/open-orchestra/open-orchestra-model-interface/pull/140
https://github.com/open-orchestra/open-orchestra-base-api-bundle/pull/65

### CODE REVIEW

File: BaseApiBundle/Controller/BaseController.php [hunk 1]
```diff
@@ -57,6 +57,7 @@ protected function reverseTransform(Request $request, $id, $type, $event, $event
         );
 
         $mixed = $this->get('open_orchestra_model.repository.' . $typeName)->find($id);
+        $oldStatus = $mixed->getStatus();
```
Comment (nicolasThal) [inline BaseApiBundle/Controller/BaseController.php]: you should check if the element is statusable

File: BaseApiBundle/Controller/BaseController.php [hunk 2]
```diff
@@ -64,7 +65,7 @@ protected function reverseTransform(Request $request, $id, $type, $event, $event
             $em->persist($mixed);
             $em->flush();
 
-            $this->dispatchEvent($event, new $eventClass($mixed));
+            $this->dispatchEvent($event, new $eventClass($mixed, $oldStatus));
```
Comment (nicolasThal) [inline BaseApiBundle/Controller/BaseController.php]: and if the event class is EventStatusable

File: BaseApiBundle/Controller/BaseController.php [hunk 3]
```diff
@@ -57,6 +58,10 @@ protected function reverseTransform(Request $request, $id, $type, $event, $event
         );
 
         $mixed = $this->get('open_orchestra_model.repository.' . $typeName)->find($id);
+        $oldStatus = null;
+        if($mixed instanceof StatusableInterface) {
```
Comment (ngilain) [inline BaseApiBundle/Controller/BaseController.php]: a little space here

File: BaseApiBundle/Controller/BaseController.php [hunk 4]
```diff
@@ -64,6 +69,11 @@ protected function reverseTransform(Request $request, $id, $type, $event, $event
             $em->persist($mixed);
             $em->flush();
 
+            if(in_array('OpenOrchestra\ModelInterface\Event\EventTrait\EventStatusable', class_uses($eventClass))) {
```
Comment (ngilain) [inline BaseApiBundle/Controller/BaseController.php]: and there too

Comment (ngilain) [inline BaseApiBundle/Controller/BaseController.php]: I'm not sure this classname should be hardcoded, it does not allow an easy extend of the code :/

Comment (veolys-nanne) [inline BaseApiBundle/Controller/BaseController.php]: It's not the name of a "class" but of a "trait". For me, the line 73 should be done only if the class 'implements' the trait. @itkg-ngilain

Comment (ngilain) [inline BaseApiBundle/Controller/BaseController.php]: @itkg-nanne : The point was about the hardcoded path. I'm not fan of that kind of thing.

**Pull request merged**